test/widgets/board_test.dart

```diff
@@ -170,6 +185,26 @@ void main() {
       expect(find.byKey(const Key('e4-lastMove')), findsOneWidget);
     });
 
+    testWidgets('Cannot move by drag if input method is tapTwoSquares', (
+      WidgetTester tester,
+    ) async {
+      await tester.pumpWidget(
+        buildBoard(
+          initialInteractableSide: InteractableSide.both,
+          inputMethod: InputMethod.tapTwoSquares,
+        ),
+      );
+      await tester.dragFrom(
+        squareOffset('e2'),
+        const Offset(0, -(squareSize * 2)),
+      );
+      await tester.pumpAndSettle();
+      expect(find.byKey(const Key('e4-whitePawn')), findsNothing);
+      expect(find.byKey(const Key('e2-whitePawn')), findsOneWidget);
+      expect(find.byKey(const Key('e2-lastMove')), findsNothing);
+      expect(find.byKey(const Key('e4-lastMove')), findsNothing);
+    });
+
```

Comment:
We could add another test for the `tapTwoSquares`: the piece is still selected after a drag attempt.

Reply:
good point, added that to the test